CDCgov/prime-reportstream
# Engagement/jessica/15540 fhirdata specify receiver

This PR adds the ability to specify a receiver on the fhirdata command so that their settings can be applied. 

Test Steps:
1. Choose a message that one of the following settings would make a change to:
truncateHDNamespaceIds,truncateHl7Fields,convertTimestampToDateTime,convertPositiveDateTimeOffsetToNegative,convertDateTimesToReceiverLocalTime,useHighPrecisionHeaderDateTimeFormat,timeZone
2. Run the fhirdata command on it example command:
`fhirdata
-i
/Users/jessicawillis/git/newnew/prime-reportstream/prime-router/src/testIntegration/resources/datatests/FHIR_to_HL7/sample_ME_20240806-0001.fhir
-o
/Users/jessicawillis/git/newnew/prime-reportstream/prime-router/src/testIntegration/resources/datatests/FHIR_to_HL7/sample_ME_20240806-0001.hl7
--output-format
HL7
-r
classpath:/metadata/hl7_mapping/receivers/STLTs/ME/ME-receiver-transform.yml
-s
classpath:/metadata/fhir_transforms/senders/SimpleReport/simple-report-sender-transform.yml
--receiver-name
full-elr
--org
mo-phd
--receiver-setting-env
staging`
3. Check the changes were made in the output file. 

## Changes
- added params to the fhirdata command.

## Checklist

### Testing

### Process

## Linked Issues
- Fixes #issue

## To Be Done
*Create GitHub issues to track the work remaining, if any*
- #issue

## Specific Security-related subjects a reviewer should pay specific attention to
- Does this PR introduce new endpoints?
    - new endpoint A
    - new endpoint B
- Does this PR include changes in authentication and/or authorization of existing endpoints?
- Does this change introduce new dependencies that need vetting?
- Does this change require changes to our infrastructure?
- Does logging contain sensitive data?
- Does this PR include or remove any sensitive information itself?

If you answered '_yes_' to any of the questions above, conduct a detailed Review that addresses at least:

- What are the potential security threats and mitigations? Please list the _STRIDE_ threats and how they are mitigated
    - **S**poofing (faking authenticity)
        - Threat _T_, which could be achieved by _A_, is mitigated by _M_
    - **T**ampering (influence or sabotage the integrity of information, data, or system)
    - **R**epudiation (the ability to dispute the origin or originator of an action)
    - **I**nformation disclosure (data made available to entities who should not have it)
    - **D**enial of service (make a resource unavailable)
    - **E**levation of Privilege (reduce restrictions that apply or gain privileges one should not have)
- Have you ensured logging does not contain sensitive data?
- Have you received any additional approvals needed for this change?
